### PR TITLE
Fix: avoid automatic device detection via serialized tensors when deserializing `PyTorchPredictor`.

### PR DESCRIPTION
**Pull request by shubhamkapoor**
*Issue #, if available:*

*Description of changes:*
When `map_location=None` is passed to `torch.load`, it automatically loads the tensors to the device they were serialized on. It results in runtime errors when the model is trained using a GPU and serialized, but inference is done on CPU when predictor is constructed by `deserialize` method of PytorchPredictor with `device=None`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Review by lostella — APPROVED** [review]
Thanks! We can then consolidate this with the same logic happening in other places, maybe having a `default_device` function to do this would be good

**Review by jaheba — APPROVED** [review]
